src/IssueViz/OpenInIde/OpenInIdeConfigScopeValidator.cs

```diff
@@ -0,0 +1,75 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.SLCore.State;
+
+namespace SonarLint.VisualStudio.IssueVisualization.OpenInIde;
+
+internal interface IOpenInIdeConfigScopeValidator
+{
+    bool TryGetConfigurationScopeRoot(string issueConfigurationScope, out string configurationScopeRoot);
+}
+
+[Export(typeof(IOpenInIdeConfigScopeValidator))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal class OpenInIdeConfigScopeValidator : IOpenInIdeConfigScopeValidator
+{
+    private readonly IActiveConfigScopeTracker activeConfigScopeTracker;
+    private readonly ILogger logger;
+    private readonly IThreadHandling threadHandling;
+
+    [ImportingConstructor]
+    public OpenInIdeConfigScopeValidator(IActiveConfigScopeTracker activeConfigScopeTracker, ILogger logger, IThreadHandling threadHandling)
+    {
+        this.activeConfigScopeTracker = activeConfigScopeTracker;
+        this.logger = logger;
+        this.threadHandling = threadHandling;
+    }
+
+    public bool TryGetConfigurationScopeRoot(string issueConfigurationScope, out string configurationScopeRoot)
+    {
+        threadHandling.ThrowIfOnUIThread();
+
+        var configScope = activeConfigScopeTracker.Current;
+        configurationScopeRoot = configScope?.RootPath;
```

Comment:
You set it first and return false but this would set the out parameter even if it's false. Are you sure it's expected behaviour?

Reply:
Yes, but I think you shouldn't also set the value when it's false.

Reply:
You have to set some value to the out parameter. There's no difference which value you set if the caller ignores the fact that the result is false.

Reply:
You should return the default value when it's false, setting something other than default is wrong behaviour.

Reply:
Why?

Reply:
Because this is the expected behaviour, If you tried to get some value and failed then you shouldn't have any value on the field. For instance how would you feel if TryGetValue of a dictionary returned a value from the dictionary even the result was false? You are creating dirty data.

Reply:
I would never use the value if the result is false

Reply:
The issue is not whether you use it or not; the issue is leaking dirty data. We should avoid this.